wrappers/javascript/nodejs/src/NodeJSAnoncreds.ts

```diff
@@ -0,0 +1,674 @@
+
+    const ret = allocatePointer()
+
+    // @ts-ignore
+    nativeAnoncreds.anoncreds_create_schema(name, version, issuerId, attributeNames, ret)
+    handleError()
+
+    return new ObjectHandle(ret.deref() as number)
+  }
+
+  public revocationRegistryDefinitionGetAttribute(options: { objectHandle: ObjectHandle; name: string }) {
+    const { objectHandle, name } = serializeArguments(options)
+
+    const ret = allocateStringBuffer()
+    nativeAnoncreds.anoncreds_revocation_registry_definition_get_attribute(objectHandle, name, ret)
+    handleError()
+
+    return ret.deref() as string
+  }
+
+  public credentialGetAttribute(options: { objectHandle: ObjectHandle; name: string }) {
+    const { objectHandle, name } = serializeArguments(options)
+
+    const ret = allocateStringBuffer()
+    nativeAnoncreds.anoncreds_credential_get_attribute(objectHandle, name, ret)
+    handleError()
+
+    return ret.deref() as string
+  }
+
+  public createCredentialDefinition(options: {
+    schemaId: string
+    schema: ObjectHandle
+    issuerId: string
+    tag: string
+    signatureType: string
+    supportRevocation: boolean
+  }): { credentialDefinition: ObjectHandle; credentialDefinitionPrivate: ObjectHandle; keyProof: ObjectHandle } {
+    const { schemaId, issuerId, schema, tag, signatureType, supportRevocation } = serializeArguments(options)
+
+    const credentialDefinitionPtr = allocatePointer()
+    const credentialDefinitionPrivatePtr = allocatePointer()
+    const keyProofPtr = allocatePointer()
+
+    nativeAnoncreds.anoncreds_create_credential_definition(
+      schemaId,
+      schema,
+      tag,
+      issuerId,
+      signatureType,
+      supportRevocation,
+      credentialDefinitionPtr,
+      credentialDefinitionPrivatePtr,
+      keyProofPtr
+    )
+    handleError()
+
+    return {
+      credentialDefinition: new ObjectHandle(credentialDefinitionPtr.deref() as number),
+      credentialDefinitionPrivate: new ObjectHandle(credentialDefinitionPrivatePtr.deref() as number),
+      keyProof: new ObjectHandle(keyProofPtr.deref() as number),
+    }
+  }
+
+  public createCredential(options: {
+    credentialDefinition: ObjectHandle
+    credentialDefinitionPrivate: ObjectHandle
+    credentialOffer: ObjectHandle
+    credentialRequest: ObjectHandle
+    attributeRawValues: Record<string, string>
+    attributeEncodedValues?: Record<string, string> | undefined
+    revocationRegistryId: string
+    revocationConfiguration?: NativeCredentialRevocationConfig | undefined
+  }): { credential: ObjectHandle; revocationRegistry: ObjectHandle; revocationDelta: ObjectHandle } {
+    const {
+      credentialDefinition,
+      credentialDefinitionPrivate,
+      credentialOffer,
+      credentialRequest,
+      revocationRegistryId,
+    } = serializeArguments(options)
+
+    const attributeNames = StringListStruct({
+      count: Object.keys(options.attributeRawValues).length,
+      // @ts-ignore
```

Comment:
Won't comment on the ts-ignores anymore. If temporary okay, but we should really avoid them. If there's a typing issue we should add a method that handles that issue and reuse that everywhere.